cricPredict/Contest/views.py

```diff
@@ -1,3 +1,239 @@
+    queryset = ''
+
+    def get(self, request, pk):
+        queryset = League.objects.filter(groups__in=[pk])
+        queryset.order_by('start_date')
+        serializer = LeagueSerializer(queryset, many=True)
+        return Response(serializer.data)
+
+
+class MatchAPIView(APIView):
+    permissions = (IsAuthenticated,)
+    serializer_class = MatchSerializer
+    queryset = ''
+
+    def post(self, request, *args, **kwargs):
+        serializer = MatchSerializer(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data)
+
+    def get(self, request, pk=None):
+        if pk is not None:
+            queryset = Match.objects.get(pk=pk)
+            serializer = ExtendedMatchSerializer(queryset)
+        else:
+            queryset = Match.objects.all()
+            queryset.order_by('time')
+            serializer = ExtendedMatchSerializer(queryset, many=True)
+
+        return Response(serializer.data)
+
+    def put(self, request, pk):
+        match = get_object_or_404(Match, pk=pk)
+        serializer = ExtendedMatchSerializer(instance=match, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(status=status.HTTP_200_OK)
+
+    def delete(self, request, pk):
+        Match.objects.filter(id=pk).delete()
+        return Response(status=status.HTTP_200_OK)
+
+    def patch(self, request, pk):
+        match = get_object_or_404(Match, pk=pk)
+        if 'winner' in request.data:
+            if not (request.data['winner'] == match.team1 or request.data['winner'] == match.team2):
+                request.data['winner'] = 'draw'
+        serializer = ExtendedMatchSerializer(instance=match, data=request.data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(status=status.HTTP_200_OK)
+
+
+class LeagueMatchesAPIView(APIView):
+    permissions = (IsAuthenticated,)
+    serializer_class = ExtendedMatchSerializer
+    queryset=''
+
+    def get(self, request, pk):
+        queryset = Match.objects.filter(league=pk)
+        queryset.order_by('time')
+        serializer = ExtendedMatchSerializer(queryset, many=True)
+        return Response(serializer.data)
+
+
+class PredictionAPIView(GenericAPIView):
+    permissions = (IsAuthenticated,)
+    serializer_class = PredictionSerializer
+    queryset = ''
+
+    def post(self, request, *args, **kwargs):
+        serializer = PredictionSerializer(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+        match = Match.objects.get(id=request.data['match'])
+        group = Group.objects.get(id=request.data['group'])
+        if not group.users.filter(id=request.data['user']).exists():
+            raise ParseError(detail="User should be the member of group")
+        if request.data['prediction'] == match.team1 or request.data['prediction'] == match.team2:
```

Comment:
we could move this logic to a commonplace (probably inside a serializer) to avoid repeating it